src/common/pf_alarm.c

```diff
@@ -998,12 +981,10 @@ static int pf_alarm_apmx_activate (pnet_t * net, pf_ar_t * p_ar)
             p_ar->apmx[ix].frame_id = PF_FRAME_ID_ALARM_HIGH;
          }
 
-         if (p_ar->apmx[ix].p_alarm_q == NULL)
-         {
-            p_ar->apmx[ix].p_alarm_q = os_mbox_create (PNET_MAX_ALARMS);
-         }
-         memset (p_ar->apmx[ix].apmr_msg, 0, sizeof (p_ar->apmx[ix].apmr_msg));
-         p_ar->apmx[ix].apmr_msg_nbr = 0;
+         /* Input alarm frame queue */
+         pf_alarm_queue_mutex_create (
```

Comment:
Does the queue need to create/destroy the mutex at runtime? This likely involves malloc/free which is best avoided except for initialization code.

Reply:
The queues are part of the big `pf_ar_t` struct, which is `memset()` on connect request from the PLC.
I think changing this should be a separate PR.